kafka/src/main/java/io/micronaut/configuration/kafka/intercept/KafkaClientIntroductionAdvice.java

```diff
@@ -653,12 +654,10 @@ private ProducerState getProducer(MethodInvocationContext<?, ?> context) {
 
             AbstractKafkaProducerConfiguration configuration;
             if (clientId != null) {
-                Optional<KafkaProducerConfiguration> namedConfig = beanContext.findBean(KafkaProducerConfiguration.class, Qualifiers.byName(clientId));
-                if (namedConfig.isPresent()) {
-                    configuration = namedConfig.get();
-                } else {
-                    configuration = beanContext.getBean(AbstractKafkaProducerConfiguration.class);
-                }
+                configuration = beanContext.findBean(KafkaProducerConfiguration.class, Qualifiers.byName(clientId))
+                    .or(() -> NameUtils.isValidHyphenatedPropertyName(clientId) ? Optional.empty() : beanContext.findBean(KafkaProducerConfiguration.class, Qualifiers.byName(NameUtils.hyphenate(clientId))))
+                    .map(AbstractKafkaProducerConfiguration.class::cast)
+                    .orElseGet(() -> beanContext.getBean(AbstractKafkaProducerConfiguration.class));
             } else {
                 configuration = beanContext.getBean(AbstractKafkaProducerConfiguration.class);
             }
```

Comment:
As above I think it might be helpful to have this extracted into a separate method that expresses the intent here. This `getProducer` method is massive, and I think any little bit like that helps to understand it better.

Reply:
Got it: 75af1cbf99787e40bbf39690e48efed5beb98c6c